### DISCUSSION TITLE
[alpha][auth][bug?] `signInWithPopup` triggers fetch of project config -> blocks popup

### DISCUSSION THREAD

**Discussion by robhogan** [firebase/firebase-js-sdk#4432] [General]
Raising this as discussion initially since I'm not 100% sure this is unique to alpha, but that's where I'm seeing it.

### Environment

  * Operating System version: macOS Big Sur (11.2)
  * Browser version: Safari (mac, desktop, *not* private) 14.0.3 (16610.4.3.1.4)
  * Firebase SDK version: 0.0.900-exp.1b15844a1
  * Firebase Product: auth (Google sign-in)

### Problem (and solution?)

I'm getting `auth/popup-blocked` errors using `signInWithPopup` with `GoogleAuthProvider` on Safari (mac desktop, *not* private) and I've traced it down to this:

https://github.com/firebase/firebase-js-sdk/blob/e8811c0424c1a0d56113abe3180eb5a2ea317a0f/packages-exp/auth-exp/src/platform_browser/popup_redirect.ts#L91

The first time `originValidation` is called it fetches project config with an API call (`identitytoolkit`, `GET v1/projects`) - apparently for no other reason than to throw with a helpful message if origins are improperly configured.

The delay caused by this fetch, which is *before* the popup is opened, causes the popup to become disassociated from the user interaction that triggered it. This causes the popup to be blocked by Safari and quite possibly others. Ideally there should be no asynchronous operations between user "click" and popup opening.

There doesn't seem to be any public API to pre-warm this project config or skip domain validation (which seems sensible in production anyway). **Commenting out that line fixes the issue with no ill effects I can see.**

**Comment by Feiyang1** [comment]
@samhorlbeck any thought?

**Reply by sam-gc** [reply]
Woops, sorry I missed this mention. I _believe_ it's fine to run this check asynchronously but I need to check with some other folks. [This comment in the Closure SDK](https://github.com/firebase/firebase-js-sdk/blob/bd1691c13a813bfb8636417987aa6acb93efedbe/packages/auth/src/iframeclient/ifchandler.js#L686) makes me think it will be fine to run this check in the background.

**Reply by sam-gc** [reply]
I've opened up this PR to address this: https://github.com/firebase/firebase-js-sdk/pull/4514

**Reply by sam-gc** [reply]
The PR is merged and should be included in the next release! Please let us know if you're still seeing issues on Safari after updating. Thanks again for the report!

**Reply by robhogan** [reply]
Thanks @samhorlbeck, will do 👍
